llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -592,9 +604,12 @@ void AIEPostRASchedStrategy::enterRegion(MachineBasicBlock *BB,
 
 void AIEPostRASchedStrategy::leaveRegion(const SUnit &ExitSU) {
   LLVM_DEBUG(dbgs() << "    << leaveRegion\n");
-  materializeMultiOpcodeInstrs();
 
   auto &BS = InterBlock.getBlockState(CurMBB);
+  if (BS.FixPoint.II) {
```

Comment:
I think I still don't understand that early exit. Is it because when trying pipelining, there is no region conflicts to solve? Could that line be changed to `BS.getStage() == Pipelining`?

Reply:
Yep that's an oversight.